libpod/image/pull.go

```diff
@@ -273,10 +275,12 @@ func (ir *Runtime) doPullImage(ctx context.Context, sc *types.SystemContext, goa
 		}
 		// If the image passed in was fully-qualified, we will have 1 refpair.  Bc the image is fq'd, we dont need to yap about registries.
 		if !goal.usedSearchRegistries {
+			if len(pullErrors) > 0 { // this should always be true
+				return nil, errors.Wrap(pullErrors[0], "unable to pull image")
+			}
 			return nil, errors.Errorf("unable to pull image, or you do not have pull access")
 		}
-		return nil, errors.Errorf("unable to find image on registries defined in %s, or you do not have pull access", registryPath)
-
+		return nil, errors.Errorf("unable to find image on registries defined in %s: %v", registryPath, pullErrors)
```

Comment:
Can you add a newline before the array of errors? Seems like it might be more readable

Reply:
> Can you add a newline before the array of errors? Seems like it might be more readable

We already vendor multierror indirectly via runtime-tools, so I've switched to using that with 1b0e789 -> 7e756ed.  The 7e756ed commit message has example `podman pull ...` output for various interesting cases.